src/rulenode_operators.jl

```diff
@@ -211,6 +211,159 @@ function _rulenode2expr(typ::Symbol, rulenode::AbstractRuleNode, grammar::Abstra
 end
 
 
+"""
+    expr2ulenode(expr::Expr, grammar::AbstractGrammar)
+
+Converts an expression into a [`AbstractRuleNode`](@ref) corresponding to the rule definitions in the grammar.
+"""
```

Comment:
This needs to be directly before the definition of `expr2rulenode(...)`. The signature in the docstring is missing an 'r', btw.